packages/osd-opensearch-archiver/src/lib/docs/generate_doc_records_stream.test.ts

```diff
@@ -30,14 +30,17 @@
 
 import sinon from 'sinon';
 import expect from '@osd/expect';
-import { delay } from 'bluebird';
 
 import { createListStream, createPromiseFromStreams, createConcatStream } from '../streams';
 
 import { createGenerateDocRecordsStream } from './generate_doc_records_stream';
 import { Progress } from '../progress';
 import { createStubStats, createStubClient } from './test_stubs';
 
+function delay(ms: number) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
```

Comment:
We have been defining this `delay` function way too often. We should move this out into an importable.

Reply:
@AMoo-Miki 
Thank you for ur review! I was wondering about that, so maybe you can tell me where I can place the `delay` function?
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3662#issuecomment-1503893712

Reply:
`packages/osd-utils` would be a perfect spot; just throw it into a new file. This way when we get rid of Node 14, we can just use `timers/promises` and delete this file.